kimchi/src/curve.rs

```diff
@@ -60,12 +60,14 @@ pub fn pallas_endos() -> &'static (
     &PALLAS_ENDOS
 }
 
-impl KimchiCurve for Affine<VestaParameters> {
-    fn sponge_params() -> &'static ArithmeticSpongeParams<Self::ScalarField> {
+impl KimchiCurve<{ mina_poseidon::pasta::ROUNDS }> for Affine<VestaParameters> {
```

Comment:
`mina_poseidon::pasta::ROUNDS` is hardcoded to 55 (you'll get a compiler error if you try to use a different number). 

Most of the places that `55` is hardcoded should also use this constant